packages/backend/src/services/UserAttributesService/UserAttributeUtils.ts

```diff
@@ -28,38 +32,90 @@ export const hasUserAttributes = (
 };
 
 export const exploreHasFilteredAttribute = (explore: Explore) =>
-    Object.values(explore.tables).some((table) =>
-        Object.values(table.dimensions).some(
-            (dimension) => dimension.requiredAttributes !== undefined,
-        ),
+    Object.values(explore.tables).some(
+        (table) =>
+            table.requiredAttributes !== undefined ||
+            Object.values(table.dimensions).some(
+                (dimension) => dimension.requiredAttributes !== undefined,
+            ),
+    );
+
+export const doesExploreMatchRequiredAttributes = (
+    explore: Explore,
+    userAttributes: UserAttributeValueMap,
+) =>
+    explore.tables[explore.baseTable].requiredAttributes === undefined ||
+    hasUserAttributes(
+        explore.tables[explore.baseTable].requiredAttributes,
+        userAttributes,
     );
-export const filterDimensionsFromExplore = (
+
+export const getFilteredExplore = (
     explore: Explore,
     userAttributes: UserAttributeValueMap,
-): Explore => ({
-    ...explore,
-    tables: Object.entries(explore.tables).reduce((at, exploreTable) => {
-        const [tableName, table] = exploreTable;
-        return {
-            ...at,
-            [tableName]: {
-                ...table,
-                dimensions: Object.entries(table.dimensions).reduce(
-                    (acc, tableDimension) => {
-                        const [dimensionName, dimension] = tableDimension;
+): Explore => {
+    if (!doesExploreMatchRequiredAttributes(explore, userAttributes)) {
+        throw new AuthorizationError(
+            "You don't have authorization to access this explore",
+        );
+    }
+
+    const filteredTableNames: string[] = Object.values(explore.tables).reduce<
+        string[]
+    >((acc, table) => {
+        if (hasUserAttributes(table.requiredAttributes, userAttributes))
+            return [...acc, table.name];
+        return acc;
+    }, []);
 
-                        if (
-                            hasUserAttributes(
-                                dimension.requiredAttributes,
-                                userAttributes,
-                            )
-                        )
-                            return { ...acc, [dimensionName]: dimension };
-                        return acc;
-                    },
-                    [],
-                ),
-            },
-        };
-    }, {}),
-});
+    return {
```

Comment:
these searches look so expensive

not for this PR but wondering how we can pre-cache some of this work. For example, if there are 10 dimensions and 5 of them required `is_admin` - then there are only two outcomes: either you're an admin and get 10 or you're not and you get 5.